c/src/include/libsbp/internal/unpacked/common.h

```diff
@@ -0,0 +1,298 @@
+  ctx->buf[ctx->offset++] = (uint8_t)((*v & 0xff000000) >> 24);
+  return true;
+}
+
+static inline size_t pack_u64(sbp_pack_ctx_t *ctx, const u64 *v) {
+  if (!SBP_CAN_PACK(ctx, u64)) { return false; }
+  ctx->buf[ctx->offset++] = (uint8_t)((*v & 0x00000000000000ff) >> 0);
+  ctx->buf[ctx->offset++] = (uint8_t)((*v & 0x000000000000ff00) >> 8);
+  ctx->buf[ctx->offset++] = (uint8_t)((*v & 0x0000000000ff0000) >> 16);
+  ctx->buf[ctx->offset++] = (uint8_t)((*v & 0x00000000ff000000) >> 24);
+  ctx->buf[ctx->offset++] = (uint8_t)((*v & 0x000000ff00000000) >> 32);
+  ctx->buf[ctx->offset++] = (uint8_t)((*v & 0x0000ff0000000000) >> 40);
+  ctx->buf[ctx->offset++] = (uint8_t)((*v & 0x00ff000000000000) >> 48);
+  ctx->buf[ctx->offset++] = (uint8_t)((*v & 0xff00000000000000) >> 56);
+  return true;
+}
+
+static inline size_t unpack_u8(sbp_unpack_ctx_t *ctx, u8 *v) {
+  if (!SBP_CAN_UNPACK(ctx, u8)) { return false; }
+  *v = ctx->buf[ctx->offset++];
+  return true;
+}
+
+static inline size_t unpack_u16(sbp_unpack_ctx_t *ctx, u16 *v) {
+  if (!SBP_CAN_UNPACK(ctx, u16)) { return false; }
+  *v = 0;
+  *v = (uint16_t)(*v | ((uint16_t)ctx->buf[ctx->offset++] << 0U));
+  *v = (uint16_t)(*v | ((uint16_t)ctx->buf[ctx->offset++] << 8U));
+  return true;
+}
+
+static inline size_t unpack_u32(sbp_unpack_ctx_t *ctx, u32 *v) {
+  if (!SBP_CAN_UNPACK(ctx, u32)) { return false; }
+  *v = 0;
+  *v |= (uint32_t)((uint32_t)ctx->buf[ctx->offset++] << 0);
+  *v |= (uint32_t)((uint32_t)ctx->buf[ctx->offset++] << 8);
+  *v |= (uint32_t)((uint32_t)ctx->buf[ctx->offset++] << 16);
+  *v |= (uint32_t)((uint32_t)ctx->buf[ctx->offset++] << 24);
+  return true;
+}
+
+static inline size_t unpack_u64(sbp_unpack_ctx_t *ctx, u64 *v) {
+  if (!SBP_CAN_UNPACK(ctx, u64)) { return false; }
+  *v = 0;
+  *v |= (uint64_t)((uint64_t)ctx->buf[ctx->offset++] << 0);
+  *v |= (uint64_t)((uint64_t)ctx->buf[ctx->offset++] << 8);
+  *v |= (uint64_t)((uint64_t)ctx->buf[ctx->offset++] << 16);
+  *v |= (uint64_t)((uint64_t)ctx->buf[ctx->offset++] << 24);
+  *v |= (uint64_t)((uint64_t)ctx->buf[ctx->offset++] << 32);
+  *v |= (uint64_t)((uint64_t)ctx->buf[ctx->offset++] << 40);
+  *v |= (uint64_t)((uint64_t)ctx->buf[ctx->offset++] << 48);
+  *v |= (uint64_t)((uint64_t)ctx->buf[ctx->offset++] << 56);
+  return true;
+}
+
+static inline size_t sbp_packed_size_s8(const s8 *v) {
+  (void)v;
+  return 1;
+}
+
+static inline size_t pack_s8(sbp_pack_ctx_t *ctx, const s8 *v) {
+  return pack_u8(ctx, (const u8 *)v);
+}
+
+static inline size_t unpack_s8(sbp_unpack_ctx_t *ctx, s8 *v) {
+  return unpack_u8(ctx, (u8 *)v);
+}
+
+static inline size_t sbp_packed_size_s16(const s16 *v) {
+  (void)v;
+  return 2;
+}
+
+static inline size_t pack_s16(sbp_pack_ctx_t *ctx, const s16 *v) {
+  return pack_u16(ctx, (const u16 *)v);
+}
+
+static inline size_t unpack_s16(sbp_unpack_ctx_t *ctx, s16 *v) {
+  return unpack_u16(ctx, (u16 *)v);
+}
+
+static inline size_t sbp_packed_size_s32(const s32 *v) {
+  (void)v;
+  return 4;
```

Comment:
Would a compiler assertion here be useful?  To validate that sizes are what we expect them to be?

Reply:
Good point. I think the size of these basic types can be added to the list of safe assumptions and a static_assert added